pkg/visor/visor.go

```diff
@@ -246,16 +245,11 @@ func (visor *Visor) Start() error {
 	}
 
 	ctx, cancel := context.WithCancel(context.Background())
+	visor.cancel = cancel
```

Comment:
Could you please split the `Start` function into multiple ones? Looks large.

Reply:
I feel this is a good point and we should deal with it in a separate ticket.